### PR TITLE
wifi: Add ext flash support for firmware patches

### PR DESCRIPTION
**Pull request by krish2718**
Pull in prerequisites to add external flash support for nRF70 firmware patches

### CODE REVIEW

File: drivers/wifi/nrfwifi/Kconfig.nrfwifi [hunk 1]
```diff
@@ -130,7 +130,8 @@ config NRF_WIFI_IF_AUTO_START
 
 config NRF_WIFI_PATCHES_BUILTIN
 	bool "Store nRF70 FW patches as part of the driver"
-	default y
+	default y if !NRF_WIFI_BUILD_ONLY_MODE
```
Comment (nordicjm) [inline drivers/wifi/nrfwifi/Kconfig.nrfwifi:133]: commit is outdated and needs updating

File: drivers/wifi/nrfwifi/rpu_fw_patches.ld
Comment (nordicjm) [inline drivers/wifi/nrfwifi/rpu_fw_patches.ld:1]: this does not match the upstream commit

**Review by nordicjm — CHANGES_REQUESTED** [review]

**Comment by krish2718** [comment]
Rebased to latest upstream after your review.

**Review by rado17 — APPROVED** [review]

File: drivers/wifi/nrfwifi/CMakeLists.txt [hunk 2]
```diff
@@ -4,7 +4,7 @@
 # SPDX-License-Identifier: Apache-2.0
 #
```
Comment (nordicjm) [inline drivers/wifi/nrfwifi/CMakeLists.txt:5]: Can find no such mention of this commit in the linked upstream PR

File: samples/net/wifi/Kconfig.sysbuild [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+config WIFI_NRF70
+	default y if BOARD_NRF7002DK_NRF5340_CPUAPP || \
+		     BOARD_NRF7002DK_NRF5340_CPUAPP_NS || \
+		     BOARD_NRF7002DK_NRF5340_CPUAPP_NRF7001 || \
+		     BOARD_NRF7002DK_NRF5340_CPUAPP_NRF7001_NS
```
Comment (nordicjm) [inline samples/net/wifi/Kconfig.sysbuild:11]: no

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: drivers/wifi/nrfwifi/CMakeLists.txt (hunk 2 above)
Comment (krish2718) [reply drivers/wifi/nrfwifi/CMakeLists.txt:5]: oops, it should be https://github.com/zephyrproject-rtos/zephyr/pull/79572 PR, will fix it thanks.

File: samples/net/wifi/Kconfig.sysbuild (hunk 3 above)
Comment (krish2718) [reply samples/net/wifi/Kconfig.sysbuild:11]: This was cherry-picked from https://github.com/nrfconnect/sdk-zephyr/pull/2072 I think I understand your concern that for customer boards the sample might not build, but I don't see an alternative?

Comment (nordicjm) [reply samples/net/wifi/Kconfig.sysbuild:11]: You set it in samples that need it, just like every other feature in zephyr and ncs does

Comment (krish2718) [reply samples/net/wifi/Kconfig.sysbuild:11]: Yes, this is the in the sample itself, am I missing anything?

Comment (nordicjm) [reply samples/net/wifi/Kconfig.sysbuild:11]: Seems I mistakenly thought it was in the sysbuild folder, my bad

File: samples/net/wifi/sample.yaml [hunk 4]
```diff
@@ -53,6 +53,7 @@ tests:
       - nrf7002dk/nrf5340/cpuapp/nrf7001
   sample.net.wifi.nrf7002ek:
     extra_args:
+      - SB_CONFIG_WIFI_NRF70=y
```
Comment (nordicjm) [inline samples/net/wifi/sample.yaml:56]: @PerMac can this instead be reworked to use the alt config root?

**Review by nordicjm — APPROVED** [review]

File: samples/net/wifi/sample.yaml (hunk 4 above)
Comment (krish2718) [reply samples/net/wifi/sample.yaml:56]: Sure, I need to checkout this `alt` feature, I will take it up in a separate PR, thanks.

**Pull request merged**